### PR TITLE
Refactor spotqrdetector class and spot interface

### PR DESCRIPTION
**Pull request by KavitShah1998**
Scope <>

- [x] Use general averaging methods from math_utils.py for find avg_spotWorld_T_marker
- [x] Remove deprecated methods like `convert_transformation_from_sophus_to_magnum` & `xyz`
- [x] Generalize SpotQRDetector to work on any camera source of spot
- [x] Add method to find transformation between 2 camera sources on spot.
- [x] For `get_avg_spotWorld_T_marker_HAND()`, make it generic by accepting `cam_id` as input .
- [x] ~Make `_get_body_T_handcam` as `_get_body_T_camframe` or something~ Remove `_get_body_T_handcam` and replace with `get_spot_a_T_b.

### CODE REVIEW

File: aria_data_loaders/aria_data_utils/aria_live_streamer.py [hunk 1]
```diff
@@ -524,13 +540,6 @@ def initialize_object_detector(self, outputs: dict = {}, object_labels: list = [
                     belongs to; smaller number means latter the segment time-wise
                 - "object_score_list" - List of Float signifying the detection score
         """
-        self.object_detector = ObjectDetectorWrapper()
```
Comment (KavitShah1998) [inline aria_data_loaders/aria_data_utils/aria_live_streamer.py:527]: @zephirefaith 

Design Qn:

Here in `initialize_object_detector` we are passing `object_labels+meta_objects` to `self.object_detector._init_object_detector`. This argument within the function is purely getting fed to Owlvit C'tor.

Then we are again explicitly setting `_core_objects` & `_meta_objects` on Line 541 & 542.

Can we just pass 2 separate arguments for object_labels & meta_object_labels to the function `self.object_detector._init_object_detector` instead?

File: aria_data_loaders/aria_data_utils/conversions.py [hunk 2]
```diff
@@ -144,6 +157,21 @@ def bd_SE3Pose_to_ros_TransformStamped(
...
+) -> SE3Pose:
+    """
+    Convert a ROS TransformStamped to BD SE3Pose
+
+    Args:
+        ...
+
+    Returns:
+        ...
+    """
```
Comment (KavitShah1998) [inline aria_data_loaders/aria_data_utils/conversions.py:171]: Fill these up.

File: aria_data_loaders/aria_data_utils/perception/april_tag_pose_estimator.py [hunk 3]
```diff
@@ -88,20 +88,17 @@ def register_marker_ids(self, marker_ids: List[int]):
             else:
                 print(f"Marker ID {marker_id} is already registered .. skipping")
 
-    def detect_markers_and_estimate_pose(
-        self, image, should_render=False
-    ) -> Tuple[object, sp.SE3]:
+    def detect_markers_and_estimate_pose(self, image) -> Tuple[object, sp.SE3]:
```
Comment (KavitShah1998) [inline aria_data_loaders/aria_data_utils/perception/april_tag_pose_estimator.py:91]: Add type to image object

File: bd_spot_wrapper/spot_wrapper/spot.py [hunk 4]
```diff
@@ -882,6 +883,20 @@ def get_magnum_Matrix4_spot_a_T_b(self, a: str, b: str, tree=None) -> mn.Matrix4
...
+        Args:
+            frame_tree_snapshot (FrameTreeSnapshot): FrameTreeSnapshot from which to extract the vision_T_body transform
+
+        Returns:
+            sophus_SE3_vision_T_body (sp.SE3): sophus SE3 Pose containing the vision_T_body transform
+        """
+        vision_tform_body = get_vision_tform_body(frame_tree_snapshot)
+        sophus_SE3_vision_T_body = bd_SE3Pose_to_sophus_SE3(bd_se3=vision_tform_body)
+        return sophus_SE3_vision_T_body
+
```
Comment (KavitShah1998) [inline bd_spot_wrapper/spot_wrapper/spot.py:899]: Try to make this generic - `get_sophus_SE3_vision_T_body(self, frame_tree_snapshot, a, b)`

**Comment by KavitShah1998** [comment]
Output comparison for comparing camera1_T_camera2

Algorithmic output : 
```
Compute camera1_T_camera2
Dist between camera1 and camera2:  0.39067275751183667
```

True output : see image to verify the distance

File: aria_data_loaders/aria_data_utils/perception/april_tag_pose_estimator.py (hunk 3 above)
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/perception/april_tag_pose_estimator.py]: added

File: aria_data_loaders/aria_data_utils/conversions.py (hunk 2 above)
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/conversions.py]: addressed

**Review by jimmytyyang — APPROVED** [review]
LGTM!

**Comment by tusharsangam** [comment]
I checked the April Tag detector class, so far the changes seem persistent with my calibration code.

File: bd_spot_wrapper/spot_wrapper/spot.py (hunk 4 above)
Comment (KavitShah1998) [reply bd_spot_wrapper/spot_wrapper/spot.py]: Updated to use `get_sophus_SE3_a_T_b(frame_tree_snapshot, rf.SPOT_WORLD_VISION, rf.SPOT_BODY)`

File: aria_data_loaders/aria_data_utils/aria_live_streamer.py (hunk 1 above)
Comment (KavitShah1998) [reply aria_data_loaders/aria_data_utils/aria_live_streamer.py:519]: ignoring this for now.

**Pull request merged**